File: src/main/java/com/github/unchama/seichiassist/effect/breaking/AbstractRoundedTask.java
package com.github.unchama.seichiassist.effect.breaking;

import org.bukkit.scheduler.BukkitRunnable;

public abstract class AbstractRoundedTask extends BukkitRunnable {
    private int round = 0;

    public abstract void firstAction();

    public abstract void secondAction();

    public void otherwiseAction() {
        cancel();
    }

    @Override
    public void run() {
        round++;
        switch (round) {
            case 1:
                firstAction();
                break;
            case 2:
                secondAction();
                break;
            default:
                otherwiseAction();
                break;
        }
    }
}

File: src/main/java/com/github/unchama/seichiassist/effect/breaking/AbstractRoundedTask.kt
package com.github.unchama.seichiassist.effect.breaking

import org.bukkit.scheduler.BukkitRunnable

abstract class AbstractRoundedTask : BukkitRunnable() {
  private var round = 0

  abstract fun firstAction()

  abstract fun secondAction()

  fun otherwiseAction() {
    cancel()
  }

  override fun run() {
    round++
    when (round) {
      1 -> firstAction()
      2 -> secondAction()
      else -> otherwiseAction()
    }
  }
}
